src/main/java/atrium/AtriumClient.java

```diff
@@ -0,0 +1,526 @@
+import java.io.OutputStream;
+import java.net.URL;
+import javax.net.ssl.HttpsURLConnection;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import com.google.gson.Gson;
+
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+
+public class AtriumClient {
+  private String environment;
+  private String mxAPIKEY;
+  private String mxCLIENTID;
+
+  // Required Parameters: environment, mxAPIKEY, mxCLIENTID
+  // Optional Parameters: None
+  public AtriumClient(String environment, String mxAPIKEY, String mxCLIENTID) {
+    this.environment = environment;
+    this.mxAPIKEY = mxAPIKEY;
+    this.mxCLIENTID = mxCLIENTID;
+  }
+
+
+  // USER
+
+
+  // Required Parameters: None
+  // Optional Parameters: identifier, isDisabled, metadata
+  public User createUser(String identifier, String isDisabled, String metadata) {
+    JsonObject inner = new JsonObject();
+    if (!identifier.equals("")) {
+      inner.addProperty("identifier", identifier);
+    }
+    isDisabled = isDisabled.toLowerCase();
+    if ((isDisabled.equals("")) || (isDisabled.equals("false"))) {
+      inner.addProperty("is_disabled", "false");
+    }
+    else {
+      inner.addProperty("is_disabled", true);
+    }
+    if (!metadata.equals("")) {
+      inner.addProperty("metadata", metadata);
+    }
+    JsonObject outer = new JsonObject();
+    outer.add("user", inner);
+    String body = outer.toString();
+
+    String response = makeRequest("POST", "/users", body);
+    return new Gson().fromJson(new Gson().fromJson(response, JsonObject.class).get("user").toString(), User.class);
+  }
+
+  // Required Parameters: userGUID
+  // Optional Parameters: None
+  public User readUser(String userGUID) {
+    String response =  makeRequest("GET", "/users/" + userGUID, "");
+    return new Gson().fromJson(new Gson().fromJson(response, JsonObject.class).get("user").toString(), User.class);
+  }
+
+  // Required Parameters: None
+  // Optional Parameters: userGUID, identifier, isDisabled, metadata
+  public User updateUser(String userGUID, String identifier, String isDisabled, String metadata) {
+    JsonObject inner = new JsonObject();
+    if (!identifier.equals("")) {
+      inner.addProperty("identifier", identifier);
+    }
+    if (!isDisabled.equals("")) {
+      inner.addProperty("is_disabled", isDisabled);
+    }
+    if (!metadata.equals("")) {
+      inner.addProperty("metadata", metadata);
+    }
+    JsonObject outer = new JsonObject();
+    outer.add("user", inner);
+    String body = outer.toString();
+
+    String response = makeRequest("PUT", "/users/" + userGUID, body);
+    return new Gson().fromJson(new Gson().fromJson(response, JsonObject.class).get("user").toString(), User.class);
+  }
+
+  // Required Parameters: None
+  // Optional Parameters: pageNumber, recordsPerPage
+  public User[] listUsers(String pageNumber, String recordsPerPage) {
+    String params = optionalParameters("", "", "", pageNumber, recordsPerPage);
```

Comment:
These calls to `optionalParameters()` are a bit hard to follow. It'd be easier to maintain and read if you just did checks for null or blank for the optional params. You could also add overloaded methods too to make it more clear which are optional or not.